src/shared/components/NotFound.tsx

```diff
@@ -129,12 +141,40 @@ const NotFoundOld: FC = () => (
   </div>
 )
 
-const NotFound: FC = () => {
-  if (isFlagEnabled('newNotFoundPage')) {
-    return <NotFoundNew />
+class NotFound extends Component<Props> {
+  componentDidMount() {
+    if (isFlagEnabled('deepLinking')) {
+      const deepLinkingMap = {
+        '/me/alerts': `/orgs/${this.props.org.id}/alerting`,
+        '/me/billing': `/orgs/${this.props.org.id}/billing`,
+        '/me/dashboards': `/orgs/${this.props.org.id}/dashboards-list`,
+        '/me/notebooks': `/orgs/${this.props.org.id}/notebooks`,
+        '/me/tasks': `/orgs/${this.props.org.id}/tasks`,
+        '/me/usage': `/orgs/${this.props.org.id}/usage`,
+      }
+
+      if (deepLinkingMap.hasOwnProperty(this.props.location.pathname)) {
+        this.props.history.replace(deepLinkingMap[this.props.location.pathname])
+        return
+      }
+    }
```

Comment:
converted this to a class component because redirecting in the render method / function component caused errors.

Reply:
yea, there's a way to handle that. You can do it in `useEffect` but it's fine if you've got it.